packages/gatsby-source-contentful/src/rich-text.js

```diff
@@ -1,185 +1,69 @@
-const _ = require(`lodash`)
-const { BLOCKS, INLINES } = require(`@contentful/rich-text-types`)
+import { documentToReactComponents } from "@contentful/rich-text-react-renderer"
+import resolveResponse from "contentful-resolve-response"
 
-const isEntryReferenceNode = node =>
-  [
-    BLOCKS.EMBEDDED_ENTRY,
-    INLINES.ENTRY_HYPERLINK,
-    INLINES.EMBEDDED_ENTRY,
-  ].indexOf(node.nodeType) >= 0
+function renderRichText({ raw, references }, options = {}) {
+  const richText = JSON.parse(raw)
 
-const isAssetReferenceNode = node =>
-  [BLOCKS.EMBEDDED_ASSET, INLINES.ASSET_HYPERLINK].indexOf(node.nodeType) >= 0
-
-const isEntryReferenceField = field => _.get(field, `sys.type`) === `Entry`
-const isAssetReferenceField = field => _.get(field, `sys.type`) === `Asset`
-
-const getFieldProps = (contentType, fieldName) =>
-  contentType.fields.find(({ id }) => id === fieldName)
-
-const getAssetWithFieldLocalesResolved = ({ asset, getField }) => {
-  return {
-    ...asset,
-    fields: _.mapValues(asset.fields, getField),
+  // If no references are given, there is no need to resolve them
+  if (!references) {
+    return documentToReactComponents(richText, options)
   }
-}
 
-const getFieldWithLocaleResolved = ({
-  field,
-  contentTypesById,
-  getField,
-  defaultLocale,
-  resolvedEntryIDs,
-}) => {
-  // If the field is itself a reference to another entry, recursively resolve
-  // that entry's field locales too.
-  if (isEntryReferenceField(field)) {
-    const key = `${field.sys.id}___${field.sys.type}`
-    if (resolvedEntryIDs.has(key)) {
-      return field
+  // We need the original id from contentful as id attribute
+  // This code can be simplified as soon we properly replicate
+  // Contentful's sys object
+  const traverse = obj => {
```

Comment:
Nit: Could move this fn to the top scope and come up with a bit clearer naming for readability.